src/libmongoc/src/mongoc/mongoc-server-api.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "mongoc-config.h"
+#include "mongoc-server-api-private.h"
+
+const char *
+mongoc_server_api_version_to_string (mongoc_server_api_version_t version)
+{
+   if (version == MONGOC_SERVER_API_V1) {
+      return "1";
+   } else {
+      return NULL;
+   }
+}
+
+bool
+mongoc_server_api_version_from_string (const char *version,
```

Comment:
Should this be public? I think the spec only requires a way to pass a string as the API version if an enum is not possible.

Reply:
Actually, I don't think the spec requires either of these helpers to be exposed.  It does seem like something a user might be interested in, but I'm also happy to make these private.  I do think it would be odd to provide an `enum -> string` method without the parallel `string -> enum` method, so we should either expose both or hide both IMO.

Reply:
PHP makes use of this: https://github.com/mongodb/mongo-php-driver/pull/1204/files#diff-4ddc908579a884b62260840125c4a168a175022f1a5f764db111cd22a0529899R35. Not having this helper exposed would require us to duplicate that logic in PHP to be able to create a `mongoc_server_api_version_t`.

Reply:
Ok, then I will leave these as-is.